src/main/scala/com/hpe/sbt/ValidatePullRequest.scala

```diff
@@ -132,20 +132,35 @@ object ValidatePullRequest extends AutoPlugin {
   private lazy val travisPullRequestId = sys.env.get(TravisPullIdEnvVarName).filterNot(_ == "false").map(_.toInt)
   private lazy val pullRequestId = jenkinsPullRequestId orElse travisPullRequestId
 
-  override lazy val buildSettings = Seq(
-    validatePullRequest / includeFilter := "*",
-    validatePullRequest / excludeFilter := "README.*",
-    validatePullRequestBuildAll / includeFilter := PathGlobFilter("project/**") || PathGlobFilter("*.sbt"),
-    validatePullRequestBuildAll / excludeFilter := NothingFilter,
+  override lazy val globalSettings = Seq(
+    validatePullRequest := (),
+    validatePullRequestBuildAll := (),
     prValidatorSourceBranch := {
       localSourceBranch orElse jenkinsSourceBranch getOrElse "HEAD"
     },
     prValidatorTargetBranch := {
       localTargetBranch orElse jenkinsTargetBranch orElse travisTargetBranch getOrElse "origin/main"
     },
+    prValidatorTasks := Nil,
+    prValidatorBuildAllTasks := Nil,
+    prValidatorEnforcedBuildAllTasks := Nil,
     prValidatorGithubEndpoint := uri("https://api.github.com"),
     prValidatorGithubRepository := sys.env.get(TravisRepoName),
     prValidatorBuildAllKeyword := """PLS BUILD ALL""".r,
+    prValidatorGithubEnforcedBuildAll := false,
+    prValidatorTravisNonPrEnforcedBuildAll := {
+      sys.env.get(TravisPullIdEnvVarName).contains("false")
+    },
+    prValidatorEnforcedBuildAll := false,
+    prValidatorChangedProjects := Changes(allBuildMatched = false, Set.empty),
+    prValidatorProjectBuildTasks := Nil
+  )
+
+  override lazy val buildSettings = Seq(
+    validatePullRequest / includeFilter := "*",
```

Comment:
@eed3si9n Should scoped settings based on tasks also be initialized in `globalSettings`?

i.e. `validatePullRequest` is a `TaskKey` that this project introduces and `includeFilter` is a `SettingKey` contained in `sbt.Keys`

Reply:
This is one weakness of task scoping. Task scoping is weakest, so often it's pointless to have a global or build-scoped task scoping for common keys like `includeFilter`. You can check this using `inspect` command, but if there's some other `includeFilter` in the subproject or `Compile`, it could end up picking that over the task-scoped one.

Reply:
In that case I think the PR is good to go unless you pickup on something else.